v2/pkg/protocols/common/interactsh/interactsh.go

```diff
@@ -343,22 +345,36 @@ func HasMatchers(op *operators.Operators) bool {
 	return false
 }
 
-func debugPrintInteraction(interaction *server.Interaction) {
+func (c *Client) debugPrintInteraction(interaction *server.Interaction) {
 	builder := &bytes.Buffer{}
 
 	switch interaction.Protocol {
 	case "dns":
 		builder.WriteString(fmt.Sprintf("[%s] Received DNS interaction (%s) from %s at %s", interaction.FullId, interaction.QType, interaction.RemoteAddress, interaction.Timestamp.Format("2006-01-02 15:04:05")))
-		builder.WriteString(fmt.Sprintf("\n-----------\nDNS Request\n-----------\n\n%s\n\n------------\nDNS Response\n------------\n\n%s\n\n", interaction.RawRequest, interaction.RawResponse))
+		if c.options.DebugRequest || c.options.Debug {
+			builder.WriteString(fmt.Sprintf("\n-----------\nDNS Request\n-----------\n\n%s\n\n", interaction.RawRequest))
```

Comment:
The common parts, or at least the string template should be extracted and reused.

e.g.
```go
def debugMessage(cliOptions, headerText, interaction) {
  if c.options.DebugRequest || c.options.Debug {
    return fmt.Sprintf("\n-----------\n%s\n-----------\n\n%s\n\n", "HTTP Request", interaction.RawRequest)
  }
}
```

Same for the response.